areas/mysql/src/AzureMcp.MySql/Options/BaseMySqlOptions.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json.Serialization;
+using AzureMcp.Core.Options;
+
+namespace AzureMcp.MySql.Options;
+
+public class BaseMySqlOptions : SubscriptionOptions
```

Comment:
I don't feel the base options class design is correct here. All the options here are marked as required but not all options classes that extend from this require `User`, `Server`, and `Database`. For example, the tool for listing servers doesn't consume any of those options.